app/services/service_catalog/provider_control_parameters.rb

```diff
@@ -0,0 +1,38 @@
+module ServiceCatalog
+  class ProviderControlParameters
+    attr_reader :data
+    def initialize(portfolio_item_id)
+      @portfolio_item_id = portfolio_item_id
+      @data = {}
+    end
+
+    def process
+      source_ref = service_offering_source_ref
+      TopologicalInventory.call do |api_instance|
+        # TODO: Temporay till we get this call in the topology service
+        projects = api_instance.list_container_projects.select { |p| p.source_id == source_ref }
+        update_project_list(projects.collect(&:name))
+        self
+      end
+    rescue StandardError => e
+      Rails.logger.error("ProviderControlParameters #{e.message}")
+      raise
+    end
+
+    private
+
+    def update_project_list(projects)
+      @data = JSON.parse(read_control_parameters)
+      @data['properties']['namespace']['enum'] = projects
+    end
+
+    def read_control_parameters
+      # TODO: This belongs in the topology service, temporarily hosting it in service portal
+      File.read(Rails.root.join("schemas", "json", "openshift_control_parameters.json"))
+    end
+
+    def service_offering_source_ref
```

Comment:
@mkanoor is there a need for a method looking up the `service_offering_source_ref`? I only see it used on line 10, seems like the one liner on line 35 would work on line 10 as well.

Reply:
@syncrou In line #13 I am in a loop filtering out unwanted source_refs. The topology API doesn't filter namespaces per source it gives us all the namespaces (projects) and we have to filter it out. So I had to use a temp variable to store it.

Reply:
I was just wondering why we needed `source_ref = service_offering_source_ref` when it appears `source_ref = PortfolioItem.find(@portfolio_item_id).service_offering_source_ref` would work. I just thought having a separate `service_offering_source_ref` method seemed overly complicated since it is only called once.

Reply:
@syncrou Please review I have removed that function